traffic_ops/traffic_ops_golang/ats/regexrevalidate.go

```diff
@@ -0,0 +1,249 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/apache/trafficcontrol/lib/go-log"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+)
+
+const DefaultMaxRevalDurationDays = 90
+const JobKeywordPurge = "PURGE"
+const HeaderCommentDateFormat = "Mon Jan 2 15:04:05 MST 2006"
+
+func GetRegexRevalidateDotConfig(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"cdn-name-or-id"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	cdnName, userErr, sysErr, errCode := getCDNNameFromNameOrID(inf.Tx.Tx, inf.Params["cdn-name-or-id"])
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+
+	regexRevalTxt, err := getRegexRevalidate(inf.Tx.Tx, cdnName)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting regex_revalidate.config text: "+err.Error()))
+		return
+	}
+	w.Header().Set("Content-Type", "text/plain")
+	w.Write([]byte(regexRevalTxt))
+}
+
+// getCDNNameFromNameOrID returns the CDN name from a parameter which may be the name or ID.
+// This also checks and verifies the existence of the given CDN, and returns an appropriate user error if it doesn't exist.
+// Returns the name, any user error, any system error, and any error code.
+func getCDNNameFromNameOrID(tx *sql.Tx, cdnNameOrID string) (string, error, error, int) {
+	if cdnID, err := strconv.Atoi(cdnNameOrID); err == nil {
+		cdnName, ok, err := dbhelpers.GetCDNNameFromID(tx, int64(cdnID))
+		if err != nil {
+			return "", nil, fmt.Errorf("getting CDN name from id %v: %v", cdnID, err), http.StatusInternalServerError
+		} else if !ok {
+			return "", errors.New("cdn not found"), nil, http.StatusNotFound
+		}
+		return string(cdnName), nil, nil, http.StatusOK
+	}
+
+	cdnName := cdnNameOrID
+	if ok, err := dbhelpers.CDNExists(cdnName, tx); err != nil {
+		return "", nil, fmt.Errorf("checking CDN name '%v' existence: %v", cdnName, err), http.StatusInternalServerError
+	} else if !ok {
+		return "", errors.New("cdn not found"), nil, http.StatusNotFound
+	}
+	return cdnName, nil, nil, http.StatusOK
+}
+
+func getRegexRevalidate(tx *sql.Tx, cdnName string) (string, error) {
+	maxDays, ok, err := getMaxDays(tx)
+	if err != nil {
+		return "", errors.New("getting max reval duration days from Parameter: " + err.Error())
+	}
+	if !ok {
+		maxDays = DefaultMaxRevalDurationDays
+		log.Warnln("No maxRevalDurationDays regex_revalidate.config Parameter found, using default %v.", maxDays)
```

Comment:
The UTs appear to be failing due to this line, should be `log.Warnf` with a trailing `"\n"`?

Reply:
Fixed.